### PR TITLE
fix: UI Breaks When Entering a Long Username in Profile Settings

### LINKED ISSUE #16297: [CAL-4162] Username update modal overflow
### Issue Summary

https://www.loom.com/share/5112a72ae42d4aae9bf42bf2aa4d20c3?sid=09aa27f8-b4f9-46b7-92be-e3a87aff80fe

### Steps to Reproduce

1. go to https://app.cal.com/settings/my-account/profile and try to change your username to a character that's really long

### PR DESCRIPTION
**Pull request by C-W-D-Harshit**
## What does this PR do?
This PR addresses a UI bug on the update username modal at [https://app.cal.com/settings/my-account/profile](https://app.cal.com/settings/my-account/profile), where entering a very long username would break the layout. The fix includes applying a character limit on the username input field and ensuring long usernames are truncated with an ellipsis to prevent UI issues.

- Fixes #16297
- Fixes CAL-4162

Loom Video: [https://www.loom.com/share/5112a72ae42d4aae9bf42bf2aa4d20c3?sid=09aa27f8-b4f9-46b7-92be-e3a87aff80fe](https://www.loom.com/share/5112a72ae42d4aae9bf42bf2aa4d20c3?sid=09aa27f8-b4f9-46b7-92be-e3a87aff80fe)

## Mandatory Tasks (DO NOT REMOVE)
- [x] I have self-reviewed the code (A decent-sized PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Go to [https://app.cal.com/settings/my-account/profile](https://app.cal.com/settings/my-account/profile).
2. Try to update the username with a long string and verify that the UI no longer breaks.
3. Check that a character limit is enforced on the username input field.
4. Verify that long usernames are correctly displayed with truncation (e.g., ellipsis) if they exceed the allowed space.

- No additional environment variables are required.
- Minimal test data: Any valid username.
- Expected result: The username should update without breaking the UI, and long usernames should be truncated with ellipses when displayed.

### CODE REVIEW

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Thanks for the PR! In addition to this, it gets all wonky when you have extra long current username, can you fix that as well @C-W-D-Harshit 

Screenshot 2024-08-21 174229

**Comment by C-W-D-Harshit** [comment]
Hi @Amit91848 I am currently working on that.

**Comment by C-W-D-Harshit** [comment]
Hi @Amit91848 I have fixed the issue.

[https://www.loom.com/share/94881202752344708a164c69a192b28a?sid=743f630b-2df3-467a-8255-2b7ba1a0be1d](https://www.loom.com/share/94881202752344708a164c69a192b28a?sid=743f630b-2df3-467a-8255-2b7ba1a0be1d)

**Review by anikdhabal — CHANGES_REQUESTED** [review]

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx [hunk 1]
```diff
@@ -159,23 +159,23 @@ const UsernameTextfield = (props: ICustomUsernameProps & Partial<React.Component
...
                   </Tooltip>
                 </div>
-                <div>
+                <div className="max-w-96 w-fit">
                   <p className="text-subtle" data-testid="new-username">
                     {t("new_username")}
                   </p>
                   <Tooltip content={inputUsernameValue}>
-                    <p className="text-emphasis mt-1 max-w-md overflow-hidden text-ellipsis">
+                    <p className="text-emphasis mt-1 w-full overflow-hidden text-ellipsis">
```
[the remark this answers is not in the document]
Comment (anikdhabal) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:178]: Screenshot 2024-08-21 184624

**Review by anikdhabal — DISMISSED** [review]

**Comment by C-W-D-Harshit** [comment]
@anikdhabal Yes you can also fix that by adding a new line, but what if someone just entered a very very long username 😅.
I think there should be some kind of character limit with zod validation in the username field.

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (C-W-D-Harshit) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx]: @anikdhabal Yes you can also fix that by adding a new line, but what if someone just entered a very very long username 😅. The issue would still persist.
I think there should be some kind of character limit with zod validation in the username field.

**Comment by anikdhabal** [comment]
> @anikdhabal Yes you can also fix that by adding a new line, but what if someone just entered a very very long username 😅. I think there should be some kind of character limit with zod validation in the username field.

It still manages and doesn’t break. What do we mean by a very, very long username? Are we thinking about one million words? 😅

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (anikdhabal) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx]: It still manages and doesn’t break. What do we mean by a very, very long username? Are we thinking about one million words? 😅

[the remark this answers is not in the document]
Comment (C-W-D-Harshit) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx]: I

> It still manages and doesn’t break. What do we mean by a very, very long username? Are we thinking about one million words? 😅

Nah mate 😅. I like your approach, it is a lot simpler. My approach was to just fix the code way it was.

**Comment by Amit91848** [comment]
Shouldn't these changes be made in `apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx` as well?

**Comment by anikdhabal** [comment]
> Shouldn't these changes be made in `apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx` as well?

Yeah, I think we need to make changes there as well. @C-W-D-Harshit pls do

**Comment by C-W-D-Harshit** [comment]
Hi @Amit91848, I have made the changes in PremiumTextfield.tsx.

**Review by anikdhabal — APPROVED** [review]
Looks good !

**Review by Amit91848 — APPROVED** [review]
LGTM! Thanks @C-W-D-Harshit

**Pull request merged**